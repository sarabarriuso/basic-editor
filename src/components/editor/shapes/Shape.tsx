import classnames from 'classnames';
import React, { useState } from 'react';

interface IDraggableShapeProps {
  onPointerDown: (e: React.PointerEvent<HTMLDivElement>) => void;
  onDragMove: (e: React.PointerEvent<HTMLDivElement>) => void;
  styleTransform: React.CSSProperties;
  isSelected: boolean;
}

const Shape: React.FC<IDraggableShapeProps> = ({
  onPointerDown,
  onDragMove,
  styleTransform,
  isSelected,
}) => {
  const DEFAULT_CLASSNAME = 'shape';

  const [isDragging, setIsDragging] = useState(false);

  const classname: string = classnames(DEFAULT_CLASSNAME, {
    [`${DEFAULT_CLASSNAME}--selected`]: isSelected === true,
  });

  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  const handlePointerDown = (e: any) => {
    setIsDragging(true);
    e.target.setPointerCapture(e.pointerId);
    onPointerDown(e);
  };

  const handlePointerUp = () => {
    setIsDragging(false);
  };

  const handlePointerMove = (e: React.PointerEvent<HTMLDivElement>) => {
    if (isDragging) onDragMove(e);
  };

  return (
    <div
      className={classname}
      onPointerDown={handlePointerDown}
      onPointerUp={handlePointerUp}
      onPointerMove={handlePointerMove}
      style={styleTransform}
    />
  );
};

export default Shape;
