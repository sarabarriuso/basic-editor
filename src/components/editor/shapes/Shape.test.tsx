import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Shape from './Shape';
// import ShapeWrapper from './shapes/ShapeWrapper';

export default describe('Shape Tests', () => {
  it('Shape renders successfully', () => {
    // Arrange
    // Act
    const tree = shallow(
      <Shape
        onPointerDown={() => {
          return;
        }}
        onDragMove={() => {
          return;
        }}
        styleTransform={{ width: 0 }}
        isSelected={false}
      />,
    );

    //Assert
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('Shape will contain selected css modifier when selected', () => {
    // Arrange
    // Act
    const tree = shallow(
      <Shape
        onPointerDown={() => {
          return;
        }}
        onDragMove={() => {
          return;
        }}
        styleTransform={{ width: 0 }}
        isSelected={true}
      />,
    );

    const shapeComp = tree.find('div');
    //Assert
    expect(toJson(tree)).toMatchSnapshot();
    expect(shapeComp.hasClass('shape--selected')).toBeTruthy();
  });
});
