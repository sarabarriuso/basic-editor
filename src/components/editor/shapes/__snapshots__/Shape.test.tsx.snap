// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shape Tests Shape renders successfully 1`] = `
<div
  className="shape"
  onPointerDown={[Function]}
  onPointerMove={[Function]}
  onPointerUp={[Function]}
  style={
    Object {
      "width": 0,
    }
  }
/>
`;

exports[`Shape Tests Shape will contain selected css modifier when selected 1`] = `
<div
  className="shape shape--selected"
  onPointerDown={[Function]}
  onPointerMove={[Function]}
  onPointerUp={[Function]}
  style={
    Object {
      "width": 0,
    }
  }
/>
`;
